opentelemetry-sdk/src/logs/error.rs

```diff
@@ -61,3 +61,22 @@ impl<T> From<PoisonError<T>> for LogError {
 #[derive(Error, Debug)]
 #[error("{0}")]
 struct Custom(String);
+
+/// Errors returned during shutdown
+#[derive(Error, Debug)]
+#[non_exhaustive]
+pub enum ShutdownError {
+    /// Mutex lock poisoning
+    #[error("mutex lock poisioning for {0}")]
+    MutexPoisoned(String),
```

Comment:
unsure if we should make these public.. There are strictly internal details, and a mutex poison is not user actionable at all.

Reply:
If we keep one uber-error and the caller wants to handle it, they are forced to handle variants that have nothing to do with the operation they are calling